omod/src/main/java/org/openmrs/module/fhir2/web/servlet/FhirRestServlet.java

```diff
@@ -62,8 +65,12 @@ protected void initialize() {
 		setDefaultResponseEncoding(EncodingEnum.JSON);
 		registerInterceptor(loggingInterceptor);
 		
-		getFhirContext().setNarrativeGenerator(new CustomThymeleafNarrativeGenerator(
-		        FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE, FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE));
+		List<String> propertyFiles = new ArrayList<>(
+		        Arrays.asList(globalPropertyService.getGlobalProperty(FhirConstants.NARRATIVES_OVERRIDE_PROPERTY_FILE, ""),
+		            FhirConstants.HAPI_NARRATIVES_PROPERTY_FILE, FhirConstants.OPENMRS_NARRATIVES_PROPERTY_FILE));
+		propertyFiles.removeAll(Arrays.asList("", null));
+		getFhirContext().setNarrativeGenerator(
+		    new CustomThymeleafNarrativeGenerator(propertyFiles.toArray(new String[propertyFiles.size()])));
```

Comment:
Refactored it this way since the [previous commit](https://github.com/openmrs/openmrs-module-fhir2/pull/252/commits/2204e655ba3a9aeae7f05855281c982923b49969) didn't had full coverage.